### ISSUE TITLE
whether the return value of crf in the call function needs to be the same In the stage of train and valid ？

### ISSUE THREAD

**Issue by colynhn** [tensorflow/addons#2586]
hello, @howl-anderson :

May i ask you some questions? Details are as follows：

file path: https://github.com/howl-anderson/addons/blob/feature/crf_layer_on_stable_release/tensorflow_addons/layers/crf.py: 
code: def call(self, inputs, mask=None, text=None, **kwargs):

the call func return value is always decoded_sequence in the train and valid phase， but in the keras-contrib (keras-contrib/keras_contrib/layers/crf.py): the call func is defined as "out = K.in_train_phase(train_output, test_output)", it is diffient in the train and valid phase, so i have doubts about whether the return value of crf in the call function needs to be the same In the stage of train and valid？

Looking forward to everyone's answers， thx.

**Comment by howl-anderson** [comment]
Thank you @bhack for mention me. H, @colynhn thank you for your question, I will do some investigation on this (due to the outdated codebase), and feedback here when it's done.

**Comment by colynhn** [comment]
H, @howl-anderson Is there any conclusion to this question? thank you.
